lisp/obsolete/ess-noweb-mode.el

```diff
@@ -1789,6 +1789,32 @@ thread."
       (ess-noweb-tangle-thread chunk-name buffer))))
                                         ;menu functions
 
+;;;###autoload
+(defun Rnw-mode ()
+  "Major mode for editing Sweave(R) source.
+See `ess-noweb-mode' and `R-mode' for more help."
+  (interactive)
+  (require 'ess-noweb);; << probably someplace else
+  (setq ess--make-local-vars-permanent t)
```

Comment:
How about adding a obsoletion warning here? Or how should we go about it?

Reply:
Anytime you load a lisp file from an obsolete directory Emacs automatically issues a warning message, so this is taken care of for us.

Reply:
Well, I'd be happy if I could switch away from ess-noweb to something that's uniformly better.
.. and I have used versions of polymode for a long time until a few months ago.
But it's just much less usable for me than ess-noweb last I tried.  Also polymode was split in submodules, but MELPA or ELPA or github all looked confusingly different and I gave up for the time being.   

Second issue: Suddenly to use ESS with R you need more than ESS.   This is a big change from a user / installation point of view.  Notably think of people using this via an ubuntu package, or via Vincent's Windows package.  Maybe there's should then be a super package that contains both ESS and the *.Rnw *.Rmd ... enabling packages ?

At the moment, in any case, I'd be horrified if I would be coerced into not having ess-noweb available in ESS anymore.

Reply:
>  I have used versions of polymode for a long time until a few months ago.

polymode has been rewritten at the end of August. Should provide a much better user experience. 

> I gave up for the time being.

You just have to do `M-x package-install poly-R`.

> I'd be horrified if I would be coerced into not having ess-noweb available in ESS anymore.

At the moment other packages are horrified because ess-noweb is a very bad citizen. It took many hours to some package maintainers to figure out that ess-noweb is their source of problems. 

In any case, ess-noweb is not going away for the foreseeable future. It's just obsoleted and not loaded by default.

Reply:
poly-R is essentially the super package that enables R with literate markdown or latex. If you install it then ESS and markdown get pulled in as dependencies.

Reply:
ESS is actually not an explicit dependency because of the plethora of ways it could be installed.

Reply:
Leaving it out of the dependency list doesn't mean that poly-R doesn't depend on it. The second line is (require 'ess), which seems pretty explicit to me. Any decent package management system will pull in ESS if you install poly-R.

Reply:
I didn't say it wasn't requiring ESS, of course it does because it needs R-mode. Maybe it's a good time to think about extracting R-mode from ESS actually. 

> Any decent package management system will pull in ESS if you install poly-R.

I don't know what "decent" package management system you have in mind but MELPA does not fetch packages just based on some requires in the source code. 

I am not entirely sure why you are bringing all this up. Are you unhappy with how poly-R is packaged? I don't really see any other way atm.

Reply:
It seems odd not to declare ESS as a dependency. Installing poly-R from MELPA currently fails because it can't compile without ESS. But it's not really any of my business since I don't work on polymode. :-)